Dima/netflix-roulette/src/Components/Search/Search.tsx

```diff
@@ -36,20 +36,39 @@ export default class Search extends React.Component<ISearchProps> {
     };
   };
 
+  componentDidUpdate(prevProps: ISearchProps) {
```

Comment:
It seems like you need additional HIstoryAction.PUSH check in componentDidUpdate . Because you update an url in upper component, so didUpdate get triggered.